lib/srv/alpnproxy/local_proxy.go

```diff
@@ -34,6 +34,10 @@ import (
 	"github.com/gravitational/teleport/lib/utils/aws"
 )
 
+// OnNewAcceptedConnectionFunc is a callback called when a new downstream
+// connection is accepted by the local proxy.
+type OnNewAcceptedConnectionFunc func(lp *LocalProxy, conn net.Conn)
```

Comment:
this version provides `LocalProxy` and `net.Conn` in the callback. if needed, I can also add a clone of `LocalProxyConfig`. Though I would suggest providing the cert from the outside.